frontend/components/chat/index.js

```diff
@@ -0,0 +1,162 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { connect } from "react-redux";
+import orderBy from "lodash/orderBy";
+import isEqual from "lodash/isEqual";
+import { helpers, validators } from "additional";
+import { appOperations } from "modules/app";
+import { chatOperations, chatSelectors } from "modules/chat";
+import { LIGHTNING_ID_LENGTH } from "config/consts";
+
+import SubHeader from "components/subheader";
+import ErrorFieldTooltip from "components/ui/error_field_tooltip";
+
+class ChatPage extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            disabled: false,
+            lightningIdError: null,
+            messages: orderBy([...this.props.messages, ...this.props.sendedMessages], "date", "asc"),
+            msgError: null,
+        };
+    }
+
+    componentDidUpdate(prevProps) {
+        const { messages, sendedMessages } = this.props;
+        if (!isEqual(prevProps.messages, messages) || !isEqual(prevProps.sendedMessages, sendedMessages)) {
+            this.formatMessages([...messages, ...sendedMessages]);
+        }
+    }
+
+    formatMessages = messages => this.setState({ messages: orderBy(messages, "date", "asc") });
+
+    handleSubmit = async (event) => {
+        event.preventDefault();
+        this.setState({ disabled: true });
+        const lightningId = this.lightningId.value.trim();
+        const msg = this.msg.value.trim();
+        const lightningIdError = this.props.validateLightning(lightningId);
+        const msgError = validators.validateRequired(msg);
+        if (msgError || lightningIdError) {
+            this.setState({ disabled: false, lightningIdError, msgError });
+            return;
+        }
+        await this.props.sendMsg(lightningId, msg);
+        this.setState({ disabled: false });
+    };
+
+    renderItem = (item) => {
+        const date = new Date(parseInt(item.date, 10));
+        const [ymd, hms] = helpers.formatDate(date).split(" ");
+        return (
+            <div className={`chat__msg chat__msg--${item.type}`} key={item.date}>
+                {item.memo}
+                <div dateTime={date} className="chat__date">
+                    <span className="date__ymd">{ymd}</span>
+                    <span className="date__hms">{hms}</span>
+                </div>
+            </div>
+        );
+    };
+
+    render() {
+        const { messages, disabled } = this.state;
+        const submitDisabled = disabled ||
+            !this.lightningId ||
+            !this.msg ||
+            (
+                (this.lightningId && !this.lightningId.value) ||
+                (this.msg && !this.msg.value)
+            );
+        return [
+            <SubHeader key={1} />,
+            <div key={2} className="contacts-page">
+                <div className="container">
+                    <div className="chat__msg-container">
```

Comment:
Chat is now common for all users. Depending on desires and plans add messages separation